Funcky/Extensions/EnumerableExtensions/Interleave.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+
+namespace Funcky.Extensions
+{
+    public static partial class EnumerableExtensions
+    {
+        public static IEnumerable<T> Interleave<T>(this IEnumerable<T> sequence, params IEnumerable<T>[] otherSequences)
```

Comment:
I'd give this a different name. Something like `InterleaveWith`

Reply:
To discuss